sql/src/test/java/io/crate/integrationtests/CopyIntegrationTest.java

```diff
@@ -693,4 +699,84 @@ public void testCopyWithGeneratedPartitionColumnThatIsPartOfPrimaryKey() throws
+            "{\"id\": 7, \"ts\": \"Juli\"}"              // <-- invalid timestamp
+        ), "data3.json", target);
+        tmpFileWithLines(Arrays.asList(
+            "foo",                                      // <-- invalid json
+            "{\"id\": 6, \"ts\": 1496275200000}"
+        ), "data4.json", target);
+
+        execute("copy t1 from ? with (shared=true) return summary", new Object[]{target.toUri().toString() + "*"});
+        String result = printedTable(response.rows());
+
+        // one of the first files should be processed without any error
+        assertThat(result, containsString("| 2| 0| {}"));
+        // one of the first files will have a duplicate key error
+        assertThat(result, containsString("| 1| 1| {A document with the same primary key exists already={count=1}}"));
+        // file `data3.json` has a invalid timestamp error
+        assertThat(result, containsString("data3.json| 1| 2| {failed to parse [ts]={count=2}}"));
+        // file `data4.json` has an invalid json item entry
+        assertThat(result, containsString("data4.json| 1| 1| {JSON parser error: "));
+    }
+
+    @Test
+    public void testCopyFromReturnSummaryWithFailedURI() throws Exception {
+        execute("create table t1 (id int primary key, ts timestamp)");
+
+        Path tmpDir = newTempDir(LifecycleScope.TEST);
+        String tmpDirStr = tmpDir.toUri().toString();
+
+        String filename = "nonexistingfile.json";
+        execute("copy t1 from ? return summary", new Object[]{tmpDirStr + filename});
+        assertThat(response.rowCount(), is((long) internalCluster().numDataNodes()));
+
+        for (Object[] row : response.rows()) {
+            assertThat((String) row[1], endsWith(filename));
+            assertThat(row[2], nullValue());
+            assertThat(row[3], nullValue());
+            assertThat(((Map<String, Object>) row[4]).keySet(), contains(containsString("(No such file or directory)")));
+        }
+
+        // with shared=true, only 1 data node must try to process the uri
+        execute("copy t1 from ? with (shared=true) return summary", new Object[]{tmpDirStr + filename});
+        assertThat(response.rowCount(), is(1L));
+
+        for (Object[] row : response.rows()) {
+            assertThat((String) row[1], endsWith(filename));
+            assertThat(row[2], nullValue());
+            assertThat(row[3], nullValue());
+            assertThat(((Map<String, Object>) row[4]).keySet(), contains(containsString("(No such file or directory)")));
+        }
+
+        // with shared=true and wildcards all nodes will try to match a file
+        filename =  "*.json";
+        execute("copy t1 from ? with (shared=true) return summary", new Object[]{tmpDirStr + filename});
+        assertThat(response.rowCount(), is((long) internalCluster().numDataNodes()));
+
+        for (Object[] row : response.rows()) {
+            assertThat((String) row[1], endsWith("*.json"));
```

Comment:
Should we also check the first row here if it is normal?

Reply:
its the node info, which is kinda random. imho its not worth